test-d/set-required-deep.ts

```diff
@@ -0,0 +1,34 @@
+import {expectType} from 'tsd';
+import type {SetRequiredDeep} from '../index';
+
+// Set nested key to required
+declare const variation1: SetRequiredDeep<{a?: number; b?: {c?: string}}, 'b.c'>;
+expectType<{a?: number; b: {c: string}}>(variation1);
```

Comment:
you haven't specified `'b' | 'b.c'`, therefore the expected behaviour should be `expectType<{a?: number; b?: {c: string}}>(variation1);` (b is still optional)

Reply:
Thanks for this remark. I actually did this on purpose, in my mind I thought that this is what we want.

Since `SetRequired<..., 'b.c'>` would mean that we want `b.c` to always be defined, hence we also need `b` to always be defined. Do you disagree ? 

The reason is that I thought it would be fastidious when having nested objects with many levels. Writing `SetRequired<..., 'a.b.${number}.c'>` is lighter than writing `SetRequired<..., 'a' | 
'a.b' | 'a.b.${number}' | 'a.b.${number}.c'>`.

What do you think ? If you're confident on making only the nested key required I can work on that again 😀